cmd/api/handler/search.go

```diff
@@ -65,94 +74,138 @@ func (handler SearchHandler) Search(c echo.Context) error {
 		return badRequestError(c, err)
 	}
 
+	var response []responses.SearchItem
+
 	switch {
 	case isAddress(req.Search):
-		if err := handler.searchAddress(c, req.Search); err != nil {
-			return internalServerError(c, err)
-		}
+		response, err = handler.searchAddress(c.Request().Context(), req.Search)
 	case hashRegexp.MatchString(req.Search):
-		if err := handler.searchHash(c, req.Search); err != nil {
-			return internalServerError(c, err)
-		}
+		response, err = handler.searchHash(c.Request().Context(), req.Search)
 	case namespaceRegexp.MatchString(req.Search):
```

Comment:
what do you think about adding support for searching by namespace id without leading zeros?